### PR TITLE
Decompiled menu_title_screen_loop and func_8008377C

### PR DESCRIPTION
**Pull request by DavidSM64**
We're now at 31.46% complete.

### CODE REVIEW

File: src/menu.c [hunk 1]
```diff
@@ -2339,8 +2339,204 @@ void menu_title_screen_init(void) {
...
+        sMenuGuiOpacity = (D_8012686C * 8) - 1;
+        func_80068508(0);
+        if (temp_f0 != 1.0f) {
+            render_texture_rectangle_scaled(&sMenuCurrDisplayList, sGameTitleTileOffsets, 160.0f, 52.0f, temp_f0, temp_f0, -2U, 1);
+        } else {
+            render_textured_rectangle(&sMenuCurrDisplayList, sGameTitleTileOffsets, 0xA0, 0x34, (u8) 0xFF, (u8) 0xFF, (u8) 0xFF, (u8) 0xFF);
+        }
+        if (is_controller_missing() == 0) {
+            i = 0; 
+            if (osTvType == 0) {
```
Comment (Ryan-Myers) [inline src/menu.c:2361]:
```suggestion
            if (osTvType == OS_TV_TYPE_PAL) {
```

File: src/menu.c [hunk 2]
```diff
@@ -2339,8 +2339,204 @@ void menu_title_screen_init(void) {
...
+
+    if (D_8012686C != 0) {
+        func_80067F2C(&sMenuCurrDisplayList, &sMenuCurrHudMat);
+        temp_f0 = (f32) D_8012686C * 0.03125f;
+        sMenuGuiOpacity = (D_8012686C * 8) - 1;
+        func_80068508(0);
+        if (temp_f0 != 1.0f) {
+            render_texture_rectangle_scaled(&sMenuCurrDisplayList, sGameTitleTileOffsets, 160.0f, 52.0f, temp_f0, temp_f0, -2U, 1);
+        } else {
+            render_textured_rectangle(&sMenuCurrDisplayList, sGameTitleTileOffsets, 0xA0, 0x34, (u8) 0xFF, (u8) 0xFF, (u8) 0xFF, (u8) 0xFF);
```
Comment (Ryan-Myers) [inline src/menu.c:2357]:
```suggestion
            render_textured_rectangle(&sMenuCurrDisplayList, sGameTitleTileOffsets, 160, 52, 255, 255, 255, 255);
```

File: src/menu.c [hunk 3]
```diff
@@ -2339,8 +2339,204 @@ void menu_title_screen_init(void) {
...
+                if (i == gTitleScreenCurrentOption) {
+                    var_a3 = (D_801263BC & 0x1F) * 0x10;
+                    if (var_a3 >= 0x100) {
+                        var_a3 = 0x1FF - var_a3;
+                    }
+                } else {
+                    var_a3 = 0;
+                }
+                set_text_colour(255, 255, 255, var_a3, sMenuGuiOpacity);
+                draw_text(&sMenuCurrDisplayList, -0x8000, var_s2, gTitleMenuStrings[i], ALIGN_MIDDLE_CENTER);
```
Comment (Ryan-Myers) [inline src/menu.c:2378]:
```suggestion
                draw_text(&sMenuCurrDisplayList, POS_CENTRED, var_s2, gTitleMenuStrings[i], ALIGN_MIDDLE_CENTER);
```

File: src/menu.c [hunk 4]
```diff
@@ -2339,8 +2339,204 @@ void menu_title_screen_init(void) {
...
+            D_801268E0 = 0;
+            D_801268DC = 0;
+            D_800DF9F4 = 0;
+            D_801263D8 = 0;
+        }
+    }
+    if (D_8012686C != 0) {
+        if (D_8012686C < 32) {
+            if (D_8012686C == 1) {
+                play_sound_global(0x16, 0);
```
Comment (Ryan-Myers) [inline src/menu.c:2459]:
```suggestion
                play_sound_global(SOUND_WHOOSH1, 0);
```

File: src/menu.c [hunk 5]
```diff
@@ -2339,8 +2339,204 @@ void menu_title_screen_init(void) {
...
+            }
+            D_8012686C += updateRate;
+            if (D_8012686C >= 32) {
+                D_8012686C = 32;
+                sp18->unk30 = 8.0f;
+                play_sound_global(0x11, 0);
```
Comment (Ryan-Myers) [inline src/menu.c:2465]:
```suggestion
                play_sound_global(SOUND_EXPLOSION, 0);
```

File: src/menu.c [hunk 6]
```diff
@@ -2339,8 +2339,204 @@ void menu_title_screen_init(void) {
...
+        s32 temp0 = gTitleScreenCurrentOption;
+        // D_80126838 = +1 when going up, and -1 when going down.
+        if ((D_80126838 < 0) && (gTitleScreenCurrentOption < 1)) {
+            gTitleScreenCurrentOption++;
+        }
+        if ((D_80126838 > 0) && (gTitleScreenCurrentOption > 0)) {
+            gTitleScreenCurrentOption--;
+        }
+        if (temp0 != gTitleScreenCurrentOption) {
+            play_sound_global(0xEB, 0 * contrIndex); // TODO: The `* contrIndex` here is a fake match.
```
Comment (FazanaJ) [inline src/menu.c:2505]:
```suggestion
            play_sound_global(SOUND_MENU_PICK2, 0 * contrIndex); // TODO: The `* contrIndex` here is a fake match.
```

File: src/menu.c [hunk 7]
```diff
@@ -2339,8 +2339,204 @@ void menu_title_screen_init(void) {
...
+            if (D_8012686C >= 32) {
+                D_8012686C = 32;
+                sp18->unk30 = 8.0f;
+                play_sound_global(0x11, 0);
+            }
+        } else {
+            if (D_80126870 < 6.0f) {
+                D_80126870 +=  sp1C;
+                if ((D_80126870 > 0.67f) && (D_801263E0 == 0)) {
+                    play_sound_global(0x105, 0);
```
Comment (Ryan-Myers) [inline src/menu.c:2471]:
```suggestion
                    play_sound_global(SOUND_VOICE_TT_DIDDY_KONG_RACING, 0);
```

File: src/menu.c [hunk 8]
```diff
@@ -2339,8 +2339,204 @@ void menu_title_screen_init(void) {
...
+                    D_801263E0 = 1;
+                } else if ((D_80126870 > 2.83f) && (D_801263E0 == 1)) {
+                    play_sound_global(0x106, 0);
```
Comment (Ryan-Myers) [inline src/menu.c:2474]:
```suggestion
                    play_sound_global(SOUND_VOICE_TT_PRESS_START, 0);
```

File: src/menu.c [hunk 9]
```diff
@@ -2339,8 +2339,204 @@ void menu_title_screen_init(void) {
...
+        if (D_801267D8[4] & (A_BUTTON | START_BUTTON)) {
+            for(contrIndex = 3; contrIndex > 0 && !(D_801267D8[contrIndex] & (A_BUTTON | START_BUTTON)); contrIndex--){}
+            set_active_player_index(contrIndex);
+            gMenuDelay = 1;
+            func_800C01D8(&sMenuTransitionFadeIn);
+            func_800C0170();
+            play_sound_global(0xEF, 0);
+        }
+    }
+    if (gMenuDelay >= 0x1F) {
```
Comment (FazanaJ) [inline src/menu.c:2516]:
```suggestion
    if (gMenuDelay > 30) {
```

File: src/menu.c [hunk 10]
```diff
@@ -2339,8 +2339,204 @@ void menu_title_screen_init(void) {
...
+        if (temp0 != gTitleScreenCurrentOption) {
+            play_sound_global(0xEB, 0 * contrIndex); // TODO: The `* contrIndex` here is a fake match.
+        }
+        if (D_801267D8[4] & (A_BUTTON | START_BUTTON)) {
+            for(contrIndex = 3; contrIndex > 0 && !(D_801267D8[contrIndex] & (A_BUTTON | START_BUTTON)); contrIndex--){}
+            set_active_player_index(contrIndex);
+            gMenuDelay = 1;
+            func_800C01D8(&sMenuTransitionFadeIn);
+            func_800C0170();
+            play_sound_global(0xEF, 0);
```
Comment (Ryan-Myers) [inline src/menu.c:2513]:
```suggestion
            play_sound_global(SOUND_SELECT2, 0);
```

File: src/menu.c [hunk 11]
```diff
@@ -2339,8 +2339,204 @@ void menu_title_screen_init(void) {
...
+                sp28 ^= 3;
+            }
+            load_level_for_menu(0x16, -1, sp28);
+            func_8008AEB4(0, NULL);
+            menu_init(3U);
+            return 0;
+        }
+        D_800DF460 = 0;
+        load_level_for_menu(0x27, -1, 0);
+        menu_init(0xCU);
```
Comment (Ryan-Myers) [inline src/menu.c:2534]:
```suggestion
        menu_init(MENU_OPTIONS);
```

**Review by Ryan-Myers — APPROVED** [review]
LGTM, just a few constants I'd suggest.

**Pull request merged**